### PR TITLE
Branch pe dry run fix bug

### LINKED ISSUE #171: [PE-D] case sensitive inputs
unable to enter inputs with caps, unable what goes wrong too

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: itskesin/ped#9

### LINKED ISSUE #204: [PE-D] Spacing limitation on command
the command will be invalid when i mistype an extra space infront or behind of the tags, since i am typing fast, spacing errors is likely to occur a lot. This error occurs to almost all commands in the app.

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: synCKun/ped#7

### LINKED ISSUE #206: [PE-D] Adding expired ingredients
i can add expired ingredients, which probably should not be the case?

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: synCKun/ped#5

### LINKED ISSUE #212: [PE-D] Does not detect similar ingredients
I am able to add the exact same ingredients to the list. However, they are shown as 3 different ingredient.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: chengTzeNing/ped#3

### LINKED ISSUE #230: [PE-D] Quantity 0 allowed in the list #182
It is strange that you can add a quantity 0 to the list, and does not delete the 0 quantity ingredient after cooking.

In fact I just realised the ingredients are used up in the cooking, because in the previous issue I did not notice there is a change in the quantity.

Labels: `severity.Medium` `type.FunctionalityBug`
original: https://github.com/Anqi-nus/ped/issues/6

### PR DESCRIPTION
**Pull request by jinfayap**
Fix PE Dry Run Bugs.

1. Fix case sensitive issue in search function from https://github.com/AY1920S2-CS2113T-M16-2/tp/issues/171
2.  Address issue with space limitation on command from https://github.com/AY1920S2-CS2113T-M16-2/tp/issues/204
3. Addressed adding expired ingredient from https://github.com/AY1920S2-CS2113T-M16-2/tp/issues/206
4. Address the detection of similar ingredients with same name (case ignored), same price and expiry and update the quantity from https://github.com/AY1920S2-CS2113T-M16-2/tp/issues/212
5. Address of 0 quantity in ingredient added to the list from https://github.com/AY1920S2-CS2113T-M16-2/tp/issues/230

### CODE REVIEW

File: src/main/java/seedu/kitchenhelper/command/AddIngredientCommand.java [hunk 1]
```diff
@@ -69,54 +79,73 @@ public void addToCategory(String category, ArrayList<Ingredient> ingredientList)
...
+     * @param ingredientName ingredient name.
+     * @param price          ingredient price.
+     * @param expiryDate     ingredient expiry date.
+     * @return true if has same name, price and expiry date, else return false.
+     */
+    public boolean hasSameNameAndPriceAndExpiry(Object i, String ingredientName, double price, String expiryDate) {
+        boolean sameName = ((Ingredient) i).getIngredientName().equalsIgnoreCase(ingredientName);
+        boolean samePrice = ((Ingredient) i).getPrice() == price;
+        boolean sameExpiry = ((Ingredient) i).getExpiryDate().equals(expiryDate);
+        return sameName && samePrice && sameExpiry;
```
Comment (alwayshuizhen) [inline src/main/java/seedu/kitchenhelper/command/AddIngredientCommand.java:128]: Clear and concise! 👍

File: src/main/java/seedu/kitchenhelper/command/AddIngredientCommand.java [hunk 2]
```diff
@@ -128,8 +157,22 @@ public void sortIngredientList(ArrayList<Ingredient> ingredientList) {
     @Override
     public CommandResult execute(ArrayList<Ingredient> ingredientList, ArrayList<Recipe> recipeList,
                                  ArrayList<Chore> choreList) {
-        addToCategory(categoryName, ingredientList);
-        kitchenLogs.info(logAddIngredient);
-        return new CommandResult(String.format(MESSAGE_SUCCESS, ingredientName, categoryName, quantity, price, expiry));
+        int index = mergeTogetherIndex(ingredientList);
+        
+        if (index != NEGATIVE_ONE) {
```
Comment (alwayshuizhen) [inline src/main/java/seedu/kitchenhelper/command/AddIngredientCommand.java:162]: Not quite sure what `NEGATIVE_ONE` is 😕 Maybe you meant `indexFound`?

**Review by alwayshuizhen — APPROVED** [review]
LGTM!

File: src/main/java/seedu/kitchenhelper/command/AddIngredientCommand.java (hunk 2 above)
Comment (jinfayap) [reply src/main/java/seedu/kitchenhelper/command/AddIngredientCommand.java:162]: Just -1. don't want use magic numbers.

**Pull request merged**